korma/src/commonMain/kotlin/com/soywiz/korma/geom/IRectangle.kt

```diff
@@ -0,0 +1,107 @@
+package com.soywiz.korma.geom
+
+interface IRectangle {
+    val x: Double
+    val y: Double
+    val width: Double
+    val height: Double
+
+    companion object {
+        inline operator fun invoke(
+            x: Double,
+            y: Double,
+            width: Double,
+            height: Double
+        ): IRectangle = Rectangle(x, y, width, height)
+
+        inline operator fun invoke(x: Float, y: Float, width: Float, height: Float): IRectangle =
+            Rectangle(x, y, width, height)
+
+        inline operator fun invoke(x: Int, y: Int, width: Int, height: Int): IRectangle =
+            Rectangle(x, y, width, height)
+
+        // Creates a rectangle from 2 points where the (x,y) is the top left point
+        // with the same width and height as the point. The 2 points provided can be
+        // in any arbitrary order, the rectangle will be created from the projected
+        // rectangle of the 2 points.
+        //
+        // Here is one example
+        // Rect XY   point1
+        // │        │
+        // ▼        ▼
+        // ┌────────┐
+        // │        │
+        // │        │
+        // └────────┘
+        // ▲
+        // │
+        // point2
+        //
+        // Here is another example
+        // point1 (Rect XY)
+        // │
+        // ▼
+        // ┌────────┐
+        // │        │
+        // │        │
+        // └────────┘
+        //          ▲
+        //          │
+        //        point2
+        operator fun invoke(point1: IPoint, point2: IPoint): Rectangle {
+            val topLeftX = minOf(point1.x, point2.x)
+            val topLeftY = minOf(point1.y, point2.y)
+            val bottomRightX = maxOf(point1.x, point2.x)
+            val bottomRightY = maxOf(point1.y, point2.y)
+            return Rectangle(topLeftX, topLeftY, bottomRightX - topLeftX, bottomRightY - topLeftY)
```

Comment:
What about renaming:

```suggestion
            val left = minOf(point1.x, point2.x)
            val top = minOf(point1.y, point2.y)
            val right = maxOf(point1.x, point2.x)
            val bottom = maxOf(point1.y, point2.y)
            return Rectangle(left, top, right - left, bottom - top)
```

So it is simpler and clearer?